developer-model-service/catalog-info.yaml

```diff
@@ -0,0 +1,2036 @@
+# Adapted from: https://raw.githubusercontent.com/benwilcock/backstage-catalogs/main/ai-asset-catalog.yml
+####################################
+# COMPONENTS
+# A Component represents a model server that's deployed, containing metadata about it
+####################################
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: developer-model-service
+  annotations:
+    backstage.io/techdocs-ref: dir:./
+  description: 'vLLM and 3scale-based model service providing models for developer tools.'
+  links:
+    - url: https://model-service.apps.domain.com
+      title: Access
+      type: website
+      icon: WebAssett
+    - url: https://ibm-granite-8b-code-instruct-vllm.apps.domain.com
+      title: API URL
+      type: website
+      icon: WebAsset
+  tags:
+    - genai
+    - ibm-granite
+    - vllm
+    - llm
+    - developer-model-service
+    - authenticated
+    - gateway
+spec:
+  type: model-server
+  owner: user:johnmcollier
+  lifecycle: production
+  providesApis:
+    - model-service-api
+  dependsOn:
+    - resource:ibm-granite-8b-code-instruct
+    - api:model-service-api
+  profile:
+    displayName: "Developer Model Service"
+---
+####################################
+# RESOURCES
+# A resource (with type `ai-model`) represents an AI model that's deployed on an associated model server
+####################################
+apiVersion: backstage.io/v1alpha1
+kind: Resource
+metadata:
+  name: ibm-granite-8b-code-instruct
+  annotations:
+    backstage.io/techdocs-ref: dir:ibm-granite-8b-code-instruct
+  description: 'IBM Granite is a decoder-only code model for code generative tasks (e.g. fixing bugs, explaining code, documenting code. Trained with code written in 116 programming languages.'
+  links:
+    - url: https://model-service.apps.domain.com
+      title: Access
+      type: website
+      icon: WebAssett
+    - url: https://ibm-granite-8b-code-instruct-vllm.apps.domain.com
+      title: API URL
+      type: website
+      icon: WebAsset
+    - url: https://huggingface.co/ibm-granite/granite-8b-code-instruct
+      title: Model Repository
+      type: website
+      icon: WebAsset
+    - url: https://huggingface.co/ibm-granite/granite-8b-code-instruct-4k/tree/main
+      title: Download Model Artifacts
+      type: website
+      icon: WebAsset
+    - url: https://www.apache.org/licenses/LICENSE-2.0
+      title: Apache-2.0 License
+      type: website
+      icon: WebAsset
+  tags:
+    - genai
+    - ibm
+    - llm
+    - granite
+    - conversational
+    - task-text-generation
+spec:
+  type: ai-model
```

Comment:
whereas this type per your google doc:

```
type | String | The type of model being stored in the catalog | MaybeCan expose instead under the model “Resource” object as part of the techdocs.
```

Reply:
So in the google doc, the `type` refers to the model type, rather than the resource `type` (which relates to the type vs. kind debate).

Since we wanted to have a well-defined, and consistent type for all models, we're not using `spec.type` to communicate the model type, but rather using models and techdocs to do that. Instead, the model type will be `ai-model` for all model types.

Reply:
The repo readme also includes the catalog structure https://github.com/redhat-ai-dev/model-catalog?tab=readme-ov-file#rhdh-ai-catalog